backend/endpoints/messages.py

```diff
@@ -1,3 +1,78 @@
 from fastapi import APIRouter
+from fastapi import APIRouter
+from fastapi import APIRouter, BackgroundTasks, HTTPException, status
+from schema.message import Message, ThreadRequest
+from schema.response import ResponseModel
+from starlette.responses import JSONResponse
+from utils.centrifugo import Events, centrifugo_client
+from utils.db import DataStorage
+from utils.room_utils import get_room
 
 router = APIRouter()
+
+MESSAGE_COLLECTION = "chat_messages"
+test_room = "61b3fb328f945e698c7eb396"
+
+@router.post(
+    "/org/{org_id}/rooms/{room_id}/sender/{sender_id}/messages",
+    response_model=ResponseModel,
+    status_code=status.HTTP_201_CREATED,
+    responses={
+        404: {"detail": "room or sender not found"},
+        424: {"detail": "ZC Core Failed"},
+    },
+)
+async def send_message(
+    org_id, room_id, sender_id, request:ThreadRequest, background_tasks: BackgroundTasks
+):
+    """Creates and sends a message from one user to another.
+    Registers a new document to the chats database collection.
+    Returns the message info and document id if message is successfully created
+    while publishing to all members of the room in the background
+    Args:
+        org_id (str): A unique identifier of an organisation
+        request: A pydantic schema that defines the message request parameters
+        room_id: A unique identifier of the room where the message is being sent to.
+    Returns:
+        HTTP_201_CREATED {new message sent}:
+        A dict containing data about the message that was created (response_output).
+            {
+                "message_id": "61696f43c4133ddga309dcf6",
+                "data": {
+                "room_id": "61b3fb328f945e698c7eb396",
+                "sender_id": "619ba4671a5f54782939d385",
+                "text": "str",
+                "reactions": [],
+                "saved_by": [],
+                "files": [],
+                "is_pinned": bool = False
+                "is_edited": bool = False
+                "created_at": "2021-10-15T19:51:41.928908Z",
+                "thread": [],
+                        }
+            }
+    Raises:
+        HTTPException [404]: Sender not in room
+        HTTPException [404]: Room does not exist
+        HTTPException [424]: "message not sent"
+    """
+    DB = DataStorage(org_id)
+    message_obj = Message(**request.dict(), org_id=org_id, room_id=room_id,
+                            sender_id= sender_id)
+    response = await DB.write(MESSAGE_COLLECTION, message_obj.dict())
+
+    if response.get("status_code") != 201:
```

Comment:
The response variable is going to be the return value of the db.write operation.
The only time it will have the key  "status_code" is actually when the request fails.

Though this works but request.get("status_code") can never be equal to 201 because db.write isn't returning the response rather it is returning response.json() on a success request. The presence of key "status_code" actually means the request fails.

therefore the condition can be rewritten as "if response and  response.get("status_code") == None"
this condition denotes a success, based on what db.write() is returning

Reply:
fixed

Reply:
`if response and response.get("status_code") is None`

Reply:
@AI-fae, please it's better to use `is` over `==` when comparing to `None`

Reply:
yea I agree with @zxenonx 
@AI-fae please kindly refactor to this

Reply:
fixed